gymnasium/envs/classic_control/cartpole.py

```diff
@@ -128,8 +129,9 @@ def __init__(self, render_mode: Optional[str] = None):
         self.steps_beyond_terminated = None
 
     def step(self, action):
-        err_msg = f"{action!r} ({type(action)}) invalid"
-        assert self.action_space.contains(action), err_msg
+        assert self.action_space.contains(
+            action
+        ), f"{action!r} ({type(action)}) invalid"
```

Comment:
I believe this causes performance issues if I remember correctly.

Reply:
The opposite - if you create `err_msg` ahead of time, then it gets created at every step. If the entire string creation is behind `assert`, then if the `assert` doesn't fire, it's basically ignored